tests/testthat/test-corpus.R

```diff
@@ -466,15 +466,54 @@ test_that("corpus handles NA correctly (#1372)", {
     ))
 })
 
-test_that("raise error when data.frame has NA in col names (#1388)", {
-    df <- data.frame(text = letters, 
-                     var1 = 1:26, 
-                     var2 = 1:26, 
-                     stringsAsFactors = F)
-    colnames(df) <- c("text", "docvar1")
-    expect_error(
-        corpus(df),
-        "columns of data.frame must be named"
+test_that("correctly handle data.frame with improper column names (#1388)", {
+    df <- data.frame(text = LETTERS[1:5],
+                     dvar1 = 1:5,
+                     dvar2 = letters[22:26],
+                     dvar3 = 6:10,
+                     stringsAsFactors = FALSE)
+    
+    # when one column name is NA
+    names(df)[3] <- NA
+    expect_equal(
+        corpus(df) %>% docvars() %>% names(),
+        c("dvar1", "V2", "dvar3")
```

Comment:
I am not sure about this expectation. I think this is more natural to make it `"dvar1", "V1", "dvar3"`, ignoring the named ones. If there column original names are `"V1", NA, "V2"`, then `"V1", "V3", "V2"`.

Or should the new column names be followed by column index?

Reply:
Ah, you just found a potential run-time error - if the original names of the data.frame were `c("V2", NA)`. So we should probably issue one more call to `make.names(names(x, unique = TRUE)`. I can add that.

That's why we should always get a second pair of 👀  for a review!

Reply:
If `c("V2", NA)` becomes `c("V2", "V1")`, there will be no name collision.

This is the proposed code.
```
l <- !nzchar(names(x)) | is.na(names(x))
name <- setdiff(paste0("V", seq(length(x))), names(x))
names(x)[l] <- head(name, sum(l))
```
However, it seems that `as.data.frame.matrx()` dose nothing with colnames when at least one of them has a name.

```r
> m <- matrix(1:4, nrow = 2)
> m
     [,1] [,2]
[1,]    1    3
[2,]    2    4

> colnames(m)[2] <- "V1"
> as.data.frame(m)
  NA V1
1  1  3
2  2  4
```